### PR TITLE
Fix closing multicast UDP channel with whitelist [5732]

### PR DESCRIPTION
**Pull request by MiguelCompany**
This probably fixes issue #235, as it is related with a hang during participant destruction.

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1933/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1893/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1903/)

File: test/unittest/transport/UDPv4Tests.cpp [hunk 1]
```diff
@@ -332,7 +332,7 @@ TEST_F(UDPv4Tests, send_to_allowed_interface)
 
         if (IsAddressDefined(locator))
         {
-            descriptor.interfaceWhiteList.emplace_back("127.0.0.1");
+            // descriptor.interfaceWhiteList.emplace_back("127.0.0.1");
```
Comment (raquelalvarezbanos) [inline test/unittest/transport/UDPv4Tests.cpp]: Can we remove this line, please?

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1940/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1900/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1910/)

**Comment by raquelalvarezbanos** [comment]
Running the tests manually:

* Mac: all tests passed

**Pull request merged**